src/main/java/com/blamejared/crafttweaker/impl/recipes/handlers/vanilla/CookingRecipeHandler.java

```diff
@@ -0,0 +1,77 @@
+package com.blamejared.crafttweaker.impl.recipes.handlers.vanilla;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.ReplacementRule;
+import com.blamejared.crafttweaker.api.util.StringUtils;
+import com.blamejared.crafttweaker.impl.item.MCItemStackMutable;
+import com.google.common.collect.ImmutableMap;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.AbstractCookingRecipe;
+import net.minecraft.item.crafting.BlastingRecipe;
+import net.minecraft.item.crafting.CampfireCookingRecipe;
+import net.minecraft.item.crafting.FurnaceRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.SmokingRecipe;
+import net.minecraft.util.ResourceLocation;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@IRecipeHandler.For(BlastingRecipe.class)
+@IRecipeHandler.For(CampfireCookingRecipe.class)
+@IRecipeHandler.For(FurnaceRecipe.class)
+@IRecipeHandler.For(SmokingRecipe.class)
+public final class CookingRecipeHandler implements IRecipeHandler<AbstractCookingRecipe> {
+    @FunctionalInterface
+    private interface CookingRecipeFactory<T extends AbstractCookingRecipe> {
+        T create(final ResourceLocation id, final String group, final Ingredient ingredient, final ItemStack result, final float experience, final int cookTime);
+    }
+    
+    private static final Map<IRecipeType<?>, Pair<String, CookingRecipeFactory<?>>> LOOKUP = ImmutableMap
+            .<IRecipeType<?>, Pair<String, CookingRecipeFactory<?>>>builder()
+            .put(IRecipeType.BLASTING, Pair.of("blastFurnace", BlastingRecipe::new))
```

Comment:
Do you really need that name here, can't you just use the `IRecipeManager#commandString` later?